### PR TITLE
 Adds script to fetch release yaml while building the Operator

### PR DESCRIPTION
**Pull request by sm43**
This adds a script as a part of `make apply` which will fetch the
fetch the release yaml for components(pipelines, triggers, ..).
this will eliminate need to each time create a pull req to add
release yaml in kodata.
This allows us to fetch versioned, latest versioned and
nightly release yamls.
we need to export `PIPELINES` and `TRIGGERS` in env or pass with
`make apply`.
By default it will fetch latest versioned release.
Examples:
PIPELINES=nightly TRIGGERS=nightly
PIPELINES=v0.25.0 TRIGGERS=v0.15.0

Signed-off-by: Shivam Mukhade <smukhade@redhat.com>

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/operator/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

Now, `make apply` and `make resolve` supports fetching release yaml for components while building the Operator. By Default, it will fetch latest versioned releases of components. To get a particular release or nightly use defined following variables in env.
Examples:
PIPELINES=nightly TRIGGERS=nightly
PIPELINES=v0.25.0 TRIGGERS=v0.15.0

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -67,9 +71,10 @@ bin/%: cmd/% FORCE
 	$Q $(GO) build -mod=vendor $(LDFLAGS) -v -o $@ ./$<
 
 .PHONY: apply
-apply: | $(KO) $(KUSTOMIZE) ; $(info $(M) ko apply on $(TARGET)) @ ## Apply config to the current cluster
-	$Q $(KUSTOMIZE) build config/$(TARGET) | $(KO) apply $(PLATFORM) -f -
-
+apply: | $(KO) $(KUSTOMIZE) ;
+	$Q ./release.sh $(TARGET) $(PIPELINES) $(TRIGGERS)|| exit ;
```
Comment (vdemeester) [inline Makefile:75]: Could we move the script in `./hack` ?

Comment (vdemeester) [reply Makefile:75]: We also need to have this for `resolve`, so most likely making this its own target and mark `apply` and `resolve` to be dependencies of it.

Comment (sm43) [reply Makefile]: yep 👍🏽

File: Makefile [hunk 2]
```diff
@@ -66,8 +70,12 @@ FORCE:
 bin/%: cmd/% FORCE
 	$Q $(GO) build -mod=vendor $(LDFLAGS) -v -o $@ ./$<
 
+.PHONY:
```
Comment (nikhil-thomas) [inline Makefile:73]:
```suggestion
.PHONY: get-releases
```

File: Makefile [hunk 3]
```diff
@@ -27,6 +27,10 @@ $(BIN)/%: | $(BIN) ; $(info $(M) building $(PACKAGE)…)
 	   rm -rf $$tmp ; exit $$ret
 
 KO = $(or ${KO_BIN},${KO_BIN},$(BIN)/ko)
+
+PIPELINES ?= latest
+TRIGGERS ?= latest
+
```
Comment (savitaashture) [inline Makefile:33]: can we do it for dashboard also ?

**Comment by savitaashture** [comment]
@sm43 It would be great if you add Release notes on how to execute script 
I see it on description but release notes would be better

File: Makefile (hunk 3 above)
Comment (pradeepitm12) [reply Makefile:33]: I think it makes sense.
We may need to take care of Kubernetes and Openshift.

Comment (sm43) [reply Makefile:33]: yep @nikhil-thomas is working on adding results and dashboard as part of this :)

**Comment by sm43** [comment]
> @sm43 It would be great if you add Release notes on how to execute script
> I see it on description but release notes would be better

@savitaashture  updated :)

**Review by pradeepitm12 — APPROVED** [review]
/lgtm

**Review by vdemeester — APPROVED** [review]

**Pull request merged**